src/data/list.js

```diff
@@ -0,0 +1,176 @@
+const mongoCollections = require('../config/mongoCollections');
+const boardData = require('./boards');
+const boards = mongoCollections.boards;
+const {ObjectId} = require('mongodb');
+const error_handler = require('../errors/error-handler'); 
+
+
+module.exports = {
+    /**
+    * Gets all lists from a given board id.
+    * @param {String} id The board's id.
+    * @returns A list of list(column) objects.
+    */
+    getAllListsInBoard: async (id) => {
+        if(!id || !error_handler.checkObjectId(id)) 
+            throw new Error("id is not valid.");
+
+        const board = await boardData.readById(id);
+        let returnList = [];
+        for(let x=0; x<board.lists.length; x++) {
+            let lists = board.lists[x];
+            lists._id = lists._id.toString();
+            returnList.push(lists);
+        }
+
+        return returnList;
+    },
+
+    /**
+    * Gets a list with a given id.
+    * @param {String} listId The lists's id.
+    * @param {String} boardId The board's id.
+    * @returns A list object, if list does not exist, throws error.
+    */
+    getListById: async (listId, boardId) => {
+        if(!listId || !error_handler.checkObjectId(listId)) 
+            throw new Error("listId is not valid.");
+
+        if(!boardId || !error_handler.checkObjectId(boardId)) 
+            throw new Error("boardId is not valid.");
+        
+        const board = await boardData.readById(boardId);
+
+        let returnList;
+        for(let y=0; y<board.lists.length; y++) {
+            let list = board.lists[y];
+            if(list._id.toString() === listId) {
+                returnList = list;
+            }
+        }
+        
+        if(!returnList) throw new Error('List not found');
+
+        return returnList;
+    },
+
+    /**
+    * Creates a new list.
+    * @param {string} listName The name of the new list.
+    * @param {String} boardId The id of the board the list is added to.
+    * @returns A board object, otherwise throws error if list wasn't added.
+    */
+    addList: async (listName, boardId) => {
+        if(!boardId || !error_handler.checkObjectId(boardId)) 
+            throw new Error("id is not valid.");
+
+        if(!listName || !error_handler.checkNonEmptyString(listName))
+            throw new Error("listName is not valid.")
+
+        const boardCollection = await boards();
+
+        let listId = new ObjectId();
+        const newList = {
+            _id: listId,
+            listName: listName.trim(),
+            cardIds: []
+        };
+
+        const updateInfo = await boardCollection.updateOne(
+            { _id: ObjectId(boardId) },
+            { $push: { lists: newList } }
+        );
+
+        if(!updateInfo.matchedCount && !updateInfo.modifiedCount) throw new Error('Update failed');
```

Comment:
Never seen this before and can't verify it's validity. You can resolve this if you're confident in it.